app/questionnaire/questionnaire_store_updater.py

```diff
@@ -339,43 +387,86 @@ def update_answers(
             self._current_question
         )
 
+        answers_updated = False
+
         for answer_id, answer_value in form_data.items():
             if answer_id not in answer_ids_for_question:
                 continue
 
-            answer_updated = self._update_answer(answer_id, list_item_id, answer_value)
-            if answer_updated:
+            if answers_updated := self._update_answer(
+                answer_id, list_item_id, answer_value
+            ):
                 self._capture_section_dependencies_for_answer(answer_id)
                 self._capture_block_dependencies_for_answer(answer_id)
 
+        if answers_updated:
+            self.capture_progress_section_dependencies()
+
+    def capture_progress_section_dependencies(self) -> None:
+        self._capture_section_dependencies_progress_value_source_for_current_block()
+        self._capture_section_dependencies_progress_value_source_for_current_section()
+
     def update_progress_for_dependent_sections(self) -> None:
         """Removes dependent blocks from the progress store and updates the progress to IN_PROGRESS.
         Section progress is not updated for the current location as it is handled by `handle_post` on block handlers.
         """
+        evaluated_dependents: list[tuple] = []
 
         self._remove_dependent_blocks_and_capture_dependent_sections()
 
-        for section in self.dependent_sections:
+        chronological_dependents = self.get_chronological_section_dependents()
+
+        for section in chronological_dependents:
             if (
                 section.section_id,
                 section.list_item_id,
             ) not in self.started_section_keys():
                 continue
 
-            is_path_complete = section.is_complete
-            if is_path_complete is None:
-                is_path_complete = self._router.is_path_complete(
-                    self._router.routing_path(
-                        section.section_id, list_item_id=section.list_item_id
-                    )
+            if (section.section_id, section.list_item_id) not in evaluated_dependents:
+                self.evaluate_dependents(
+                    section.section_id,
+                    section.list_item_id,
+                    section.is_complete,
+                    evaluated_dependents,
                 )
```

Comment:
After this you need to keep track of current section, so you need to update evaluated_dependents

```py
evaluated_dependents.append((section.section_id, section.list_item_id))
```

Reply:
Added 👍